genesis/parsing/accountsParser.go

```diff
@@ -275,7 +303,189 @@ func (ap *accountsParser) GetInitialAccountsForDelegated(addressBytes []byte) []
+	shardIDs := make([]uint32, shardCoordinator.NumberOfShards()+1)
+	for i := uint32(0); i < shardCoordinator.NumberOfShards(); i++ {
+		shardIDs[i] = i
+	}
+	shardIDs[shardCoordinator.NumberOfShards()] = core.MetachainShardId
+
+	return shardIDs
+}
+
+func createMiniBlocks(shardIDs []uint32, blockType block.Type) []*block.MiniBlock {
+	miniBlocks := make([]*block.MiniBlock, 0)
+
+	for _, i := range shardIDs {
+		for _, j := range shardIDs {
+			miniBlock := &block.MiniBlock{
+				TxHashes:        nil,
+				ReceiverShardID: i,
+				SenderShardID:   j,
+				Type:            blockType,
+			}
+
+			miniBlocks = append(miniBlocks, miniBlock)
+		}
+	}
+
+	return miniBlocks
+}
+
+func (ap *accountsParser) getAllTxs(
+	indexingData map[uint32]*genesis.IndexingData,
+	mintTxs []coreData.TransactionHandler,
+) []coreData.TransactionHandler {
+	allTxs := make([]coreData.TransactionHandler, 0)
+
+	for _, txs := range indexingData {
+		allTxs = append(allTxs, txs.DelegationTxs...)
+		allTxs = append(allTxs, txs.StakingTxs...)
+		allTxs = append(allTxs, txs.DeploySystemScTxs...)
+		allTxs = append(allTxs, txs.DeployInitialScTxs...)
+	}
+
+	allTxs = append(allTxs, mintTxs...)
+
+	return allTxs
+}
+
+func (ap *accountsParser) setScrsTxsPool(
+	shardCoordinator sharding.Coordinator,
+	indexingData map[uint32]*genesis.IndexingData,
+	txsPoolPerShard map[uint32]*indexer.Pool,
+) {
+	for _, id := range indexingData {
+		for txHash, tx := range id.ScrsTxs {
+			senderShardID := shardCoordinator.ComputeId(tx.GetSndAddr())
+			receiverShardID := shardCoordinator.ComputeId(tx.GetRcvAddr())
+
+			scrTx, ok := tx.(*scrData.SmartContractResult)
+			if !ok {
+				continue
+			}
+			scrTx.GasLimit = uint64(0)
+
+			txsPoolPerShard[senderShardID].Scrs[txHash] = scrTx
+			txsPoolPerShard[receiverShardID].Scrs[txHash] = scrTx
+		}
+	}
+}
+
+func (ap *accountsParser) setTxsPoolAndMiniBlocks(
+	shardCoordinator sharding.Coordinator,
+	allTxs []coreData.TransactionHandler,
+	txsPoolPerShard map[uint32]*indexer.Pool,
+	miniBlocks []*block.MiniBlock,
+) error {
+	var senderShardID uint32
+
+	for _, txHandler := range allTxs {
+		receiverShardID := shardCoordinator.ComputeId(txHandler.GetRcvAddr())
+
+		// treat minting transactions as intra shard
+		if bytes.Equal(txHandler.GetSndAddr(), ap.minterAddressBytes) {
+			senderShardID = receiverShardID
+		} else {
+			senderShardID = shardCoordinator.ComputeId(txHandler.GetSndAddr())
+		}
+
+		txHash, err := core.CalculateHash(ap.marshalizer, ap.hasher, txHandler)
+		if err != nil {
+			return err
+		}
+
+		tx, ok := txHandler.(*transactionData.Transaction)
+		if !ok {
+			continue
+		}
+		tx.Signature = []byte(common.GenesisTxSignatureString)
+		tx.GasLimit = uint64(0)
+
+		txsPoolPerShard[senderShardID].Txs[string(txHash)] = tx
+		txsPoolPerShard[receiverShardID].Txs[string(txHash)] = tx
+
+		for _, miniBlock := range miniBlocks {
+			if (senderShardID == miniBlock.GetSenderShardID()) &&
```

Comment:
could have used a bool var. The parenthesis can be removed

Reply:
done